tools/ClangPlugin.h

```diff
@@ -31,19 +31,25 @@ namespace clang {
 
 namespace clad {
   struct DiffRequest;
+  class EstimationPlugin;
   namespace plugin {
-    struct DifferentiationOptions {
-      DifferentiationOptions()
+  /// Register any custom error estimation model a user provides
+  using ErrorEstimationModelRegistry = llvm::Registry<EstimationPlugin>;
+  struct DifferentiationOptions {
+    DifferentiationOptions()
         : DumpSourceFn(false), DumpSourceFnAST(false), DumpDerivedFn(false),
           DumpDerivedAST(false), GenerateSourceFile(false),
-          ValidateClangVersion(false) { }
+          ValidateClangVersion(false), CustomEstimationModel(false),
+          CustomModel("") {}
 
-      bool DumpSourceFn : 1;
-      bool DumpSourceFnAST : 1;
-      bool DumpDerivedFn : 1;
-      bool DumpDerivedAST : 1;
-      bool GenerateSourceFile : 1;
-      bool ValidateClangVersion : 1;
+    bool DumpSourceFn : 1;
+    bool DumpSourceFnAST : 1;
+    bool DumpDerivedFn : 1;
+    bool DumpDerivedAST : 1;
+    bool GenerateSourceFile : 1;
+    bool ValidateClangVersion : 1;
+    bool CustomEstimationModel : 1;
+    std::string CustomModel;
```

Comment:
```suggestion
    std::string CustomModelName;
```
To maybe avoid confusion with the other `CustomModel`.